Bottom: 6c716ab2c45ddaf130f41ef97d7926eb37230b02
Top:    8d7a46c2e06e9ceb9cbdf890dc56611fb2f54ebe
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-12-24 02:25:47 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index 8f4434e..87c5cfb 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2441,7 +2441,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
